src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Services/NuGetProjectManagerService.cs

```diff
@@ -343,7 +343,7 @@ public async ValueTask ExecuteActionsAsync(IReadOnlyList<ProjectAction> actions,
                 var projectAction = new ProjectAction(
                     CreateProjectActionId(),
                     projectId,
-                    packageIdentity,
+                    resolvedAction.Action.PackageIdentity,
```

Comment:
This is the primary fix.  The test for it is [here](https://github.com/NuGet/NuGet.Client/pull/3645/files#diff-0c1b01280c4ac523b1c8e3d6fe76cad9R240).